Update TOML Parser to 1.1.0 specification

matthiasblaesing opened the pull request:
- trailing commas in arrays are now explicitly allowed (https://toml.io/en/v1.1.0#array)
- trailing commas in inline table declarations are explicitly allowed and can span multiple lines (https://toml.io/en/v1.1.0#inline-table)
- test and build dependencies were bumped (not the runtime dependency for antlr)
- javac release flag was activated to ensure code usage matches target classpath (this showed, that the source already targetted JDK 11, as `String#isBlank` is used)

matthiasblaesing committed:
Update toml grammar to align with 1.1.0 specification

- trailing commas in array and inline table declarations are explicitly
  allowed
- inline tables can span multiple lines

matthiasblaesing committed:
Bump dependencies (test dependencies and build plugins)

matthiasblaesing committed:
Use javac release-flag to compile for correct classpath (String#isBlank is used => minimum JDK is 11)

matthiasblaesing commented:
~Just noticed a problem - returning to draft.~

False alarm - the problem was in the cargo project parser from NetBeans. That had recursively added the current directory. The TOML parser was just the tip of the stack, that was visible in the traces.

vieiro commented:
Hi @matthiasblaesing ,
Please allow me some more days to review this PR. Thanks!

matthiasblaesing commented:
@vieiro no problem.

The failing run was caused by junit version upgrades, that are not JDK 11 compatible (I ran with JDK 21 ...). I updated in place and changed updates for junit to 5.14.1 which built/tested successfully for me on JDK 11.

vieiro commented:
Hi @matthiasblaesing . I'll be very busy during the following two weeks, so this will take some time. Please ping me here if this is urgent.

matthiasblaesing commented:
@vieiro no worries! There is no rush.

vieiro reviewed (approved):
Hi @matthiasblaesing .
Thanks for updating to TOML v 1.1 and apologies for the delay.
The changes look correct to me.
I'll bump the version in subsequent commits, and then publish the updated version.

matthiasblaesing commented:
@vieiro thanks for taking care. I noticed the javadoc publishing problem, that should be fixed by #21.
